django_socio_grpc/tests/test_run_server.py

```diff
@@ -0,0 +1,243 @@
+        fake_async_aio_server = FakeAsyncAioServer()
+        grpc_aio_server_mock.return_value = fake_async_aio_server
+
+        # launch command
+        call_command("grpcrunaioserver")
+
+        # assert handler function launched
+        grpc_settings.ROOT_HANDLERS_HOOK.assert_called_with(fake_async_aio_server)
+
+        # Verify that GRPC_ASYNC settings is correctly set to True
+        self.assertTrue(grpc_settings.GRPC_ASYNC)
+
+    @override_settings(
+        GRPC_FRAMEWORK={
+            **settings.GRPC_FRAMEWORK,
+            "ROOT_HANDLERS_HOOK": mock.AsyncMock(),
+            "SERVER_INTERCEPTORS": ["FakeInterceptor"],
+            "SERVER_OPTIONS": [
+                ("grpc.max_metadata_size", 1024 * 1024),
+                ("grpc.max_send_message_length", 100 * 1024 * 1024),
+                ("grpc.max_receive_message_length", 100 * 1024 * 1024),
+            ],
+        }
+    )
+    @mock.patch("grpc.aio.server")
+    def test_server_launch_with_interceptor_and_options(self, grpc_aio_server_mock):
+        """
+        Testing that the grpc server is lauching in its more easy way.
+        It verify that all the need grpc methods are correctly called
+        """
+        # Mocking
+        fake_async_aio_server = FakeAsyncAioServer()
+        grpc_aio_server_mock.return_value = fake_async_aio_server
+
+        # launch command
+        call_command("grpcrunaioserver")
+
+        # assert grpc server launch with corrects args
+        grpc_aio_server_mock.assert_called_with(
+            mock.ANY,
+            interceptors=["FakeInterceptor"],
+            options=[
+                ("grpc.max_metadata_size", 1048576),
+                ("grpc.max_send_message_length", 104857600),
+                ("grpc.max_receive_message_length", 104857600),
+            ],
+        )
+
+    @override_settings(
+        GRPC_FRAMEWORK={
+            **settings.GRPC_FRAMEWORK,
+            "ROOT_HANDLERS_HOOK": mock.AsyncMock(),
+            "PRIVATE_KEY_CERTIFICATE_CHAIN_PAIRS_PATH": [["server-key.pem", "server.pem"]],
+        }
+    )
+    @mock.patch("grpc.aio.server")
+    @mock.patch("grpc.ssl_server_credentials")
+    def test_secure_connection(self, mock_ssl_server_credentials, grpc_aio_server_mock):
+        """
+        Testing that if PRIVATE_KEY_CERTIFICATE_CHAIN_PAIRS_PATH correctly set then we launch a secure server
+        """
+
+        # Mocking
+        fake_async_aio_server = FakeAsyncAioServer()
+        grpc_aio_server_mock.return_value = fake_async_aio_server
+
+        server_credentials_fake_return_value = "Fake"
+        mock_ssl_server_credentials.return_value = server_credentials_fake_return_value
+
+        read_file_mock_data = "fakefiledata"
+
+        with patch_open(read_data=read_file_mock_data) as mocked_open:
+            # launch command
+            call_command("grpcrunaioserver")
+
+        # verfiy open mock to be sure cert file are correctly read and passed to ssl_server_credentials
```

Comment:
you're testing read internals, i don't think it's relevant when you already test the output below

Reply:
As a mock open in global way I want to be sur the mock it's called with the correct args to be sure step are not inversed